bikram.md

```diff
@@ -0,0 +1,95 @@
+There are two ways to save authentication information in the browser:
+	Cookies
+	HTML5 Web Storage
+In each case, you have to trust that browsers are implemented correctly, and that Website A can't somehow access the authentication information for Website B. In that sense, both storage mechanisms are equally secure. Problems can arise in terms of how you use them though.
+•	If you use cookies: The browser will automatically send the authentication information with every request to the API. This can be convenient so long as you know it's happening. Prevention against CSRF is a must while using this technique. It is also strongly recommended to use cookies with HTTPOnly and Secure flags set. This will allow the browser to send along the token for authentication purposes, but won’t expose it to the JavaScript environment. 
+•	If you use HTML5 Web Storage: You have to write JavaScript that manages exactly when and what authentication information is sent.
+
+11	Protect against Cross-Site Request Forgery:
+For resources exposed by RESTful web services, it's important to make sure any PUT, POST, and DELETE request is protected from Cross Site Request Forgery. Typically, one would use a token-based approach. See Cross-Site Request Forgery (CSRF) Prevention Cheat Sheet for more information on how to implement CSRF-protection. CSRF is easily achieved even using random tokens if any XSS exists within your application, so please make sure you understand how to prevent XSS.
+
+12	Insecure direct object references:
+A URL or even a POSTed form should NEVER contain an access control "key" or similar that provides automatic verification. A data contextual check needs to be done, server side, with each request.
+
+13	Enable CORS for all APIs:
+When your API's resources receive requests from a domain other than the API's own domain, you must enable cross-origin resource sharing (CORS) for selected methods on the resource. This amounts to having your API respond to the OPTIONS preflight request with at least the following CORS-required response headers:
+•	Access-Control-Allow-Methods
+•	Access-Control-Allow-Headers
+•	Access-Control-Allow-Origin
+
+14	Data in transit:
+Unless the public information is completely read-only, the use of TLS v1.2 should be mandated, particularly where credentials, updates, deletions, and any value transactions are performed. The overhead of TLS is negligible on modern hardware, with a minor latency increase that is more than compensated by safety for the end user.
+
+15	Message Integrity:
+In addition to HTTPS/TLS, JSON Web Token (JWT) is an open standard that defines a compact and self-contained way for securely transmitting information between parties as a JSON object. JWT can not only be used to ensure the message integrity but also authentication of both message sender/receiver. The JWT includes the digital signature hash value of the message body to ensure the message integrity during the transmission.
```

Comment:
so the suggestion is that an API SHOULD use JWT tokens for authentication and verification of message integrity?